packages/@orchidui-vue/src/Form/Input/OcInput.vue

```diff
@@ -68,24 +68,46 @@ const props = defineProps({
     type: Object,
     default: () => ({}),
   },
+  variant: {
+    type: String,
+  },
 });
 
-defineEmits({
-  "update:modelValue": [],
-  blur: [],
-  focus: [],
-});
+defineEmits(["update:modelValue", "blur", "focus"]);
 
 const attrs = useAttrs();
 
 const eventListeners = pickEventListeners(attrs);
 
 const inputRef = ref();
+const formattedValue = ref();
 
 defineExpose({
   focus: () => inputRef.value.focus(),
 });
 
+watch(formattedValue, (data) => {
+  formattedValue.value = formatValue(data);
+})
+
+const formatValue = (value) => {
+  let output = value;
+
+  if(props.variant === 'currency') {
+    // removing non-digit characters
+    output = +(`${output}`.replace(/\D/g, ''));
+    output = (output / 100).toFixed(2);
+  }
+
+  return output;
+}
+
+onMounted(() => {
+  if(props.variant === 'currency') {
+    formattedValue.value = formatValue(props.modelValue);
+  }
+})
```

Comment:
`formatValue` should be passed by prop
```suggestion
const formattedValue = computed(() => {
  if (typeof props.formatValue === 'function') {
    return props.formatValue(props.modelValue);
  }
  
  return props.modelValue
})
```